### ISSUE TITLE
Validation with conditionally hidden field doesn't work with the sometimes option

### ISSUE THREAD

**Issue by helloDanuk** [statamic/cms#10298]
### Bug description

According to the [best practice tipp](https://statamic.dev/validation#validating-nestable-fields) I have two fields in my blueprint. The second field should only be required if the toggle is set to true.

```
  -
    handle: google_maps_link_toggle
    field:
      inline_label: Nein
      inline_label_when_true: Ja
      type: toggle
      display: 'Google Maps Link?'
      width: 33
      listable: true
  -
    handle: google_maps_link
    field:
      placeholder: 'https://goo.gl/maps/2Lpaoj3kPw8Yup2aA'
      display: 'Google Maps Link'
      type: text
      instructions: '[Google Maps](https://www.google.ch/maps) öffnen, Adresse eingeben und über den Teilen Button den Link im Format wie unten kopieren.'
      validate:
        - sometimes
        - required
      always_save: true
      if:
        google_maps_link_toggle: 'equals true'
```

But saving the entry leads to an error.

<img width="1545" alt="Bildschirmfoto 2024-06-13 um 22 30 00" src="https://github.com/statamic/cms/assets/8265902/452ea321-707e-482a-af05-1405e0588777">

If I set the toggle to true, I can see that the error comes from the conditionally hidden field.

<img width="1461" alt="Bildschirmfoto 2024-06-13 um 22 30 20" src="https://github.com/statamic/cms/assets/8265902/fb80f749-368c-40a9-afc7-0765dfbdc4a6">

I couldn't get the validation to work with `validate: 'required_with:{this}.google_maps_link_toggle'` too. Am i missing something or is this a bug?

### How to reproduce

- Create a toggle field
- Create a text field which is sometimes required, but is conditionally hidden with the toggle above
- Try to save the entry with the field conditionally hidden

### Environment

```yaml
Environment
Application Name: Schnätzi CMS
Laravel Version: 11.10.0
PHP Version: 8.3.7
Composer Version: 2.7.6
Environment: local
Debug Mode: ENABLED
URL: /
Maintenance Mode: OFF

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: CACHED

Drivers
Broadcasting: null
Cache: statamic
Database: sqlite
Logs: stack / single
Mail: smtp
Queue: sync
Session: file

Statamic
Addons: 5
Sites: 1
Stache Watcher: Enabled
Static Caching: Disabled
Version: 5.7.3 PRO

Statamic Addons
aryehraber/statamic-color-extractor: 1.5.0
rias/statamic-color-swatches: 2.3.0
rias/statamic-redirect: 3.7.1
statamic/seo-pro: 6.0.3
swiftmade/statamic-clear-assets: 3.0.0
```

### Installation

Fresh statamic/statamic site via CLI

**Comment by robdekort** [comment]
Hey, this seems a duplicate of #9334, check this comment for an explanation: https://github.com/statamic/cms/issues/9334#issuecomment-1904973108

TLDR: remove `always_save: true`.

**Comment by helloDanuk** [comment]
Thanks for the tip!
